pep-0692.rst

```diff
@@ -0,0 +1,594 @@
+``kwds`` rule in `the grammar <https://docs.python.org/3/reference/grammar.html>`__ 
+as follows:
+
+Before:
+
+.. code-block:: peg
+
+    kwds: '**' param_no_default 
+
+After:
+
+.. code-block:: peg
+
+    kwds:
+        | '**' param_no_default_double_star_annotation
+        | '**' param_no_default
+
+    param_no_default_double_star_annotation:
+        | param_double_star_annotation & ')'
+
+    param_double_star_annotation: NAME double_star_annotation
+
+    double_star_annotation: ':' double_star_expression
+
+    double_star_expression: '**' expression
+
+A new AST node needs to be created so that type checkers can differentiate the
+semantics of the new syntax from the existing one, which indicates that all
+``**kwargs`` should be of the same type. Then, whenever the new syntax is
+used, type checkers will be able to take into account that ``**kwargs`` should
+be unpacked. The proposition is to add a new ``DoubleStarred`` AST node. Then,
+an AST node for the function defined as::
+
+    def foo(**kwargs: **Movie): ...
+
+should look as below::
+
+    FunctionDef(
+      name='foo',
+      args=arguments(
+        posonlyargs=[],
+        args=[],
+        kwonlyargs=[],
+        kw_defaults=[],
+        kwarg=arg(
+          arg='kwargs',
+          annotation=DoubleStarred(
+            value=Name(id='Movie', ctx=Load()),
+            ctx=Load())),
+        defaults=[]),
+      body=[
+        Expr(
+          value=Constant(value=Ellipsis))],
+      decorator_list=[])
+
+The runtime annotations should be consistent with the AST. Continuing the
+previous example::
+
+    >>> def foo(**kwargs: **Movie): ...
+    ...
+    >>> foo.__annotations__
+    {'kwargs': **Movie}
+
+The double asterisk syntax should call the ``__unpack__`` special method on
+the object it was used on. This means that ``def foo(**kwargs: **T): ...`` is
+equivalent to ``def foo(**kwargs: T.__unpack__()): ...``. In addition,
+``**Movie`` in the example above is the ``repr`` of the object that
+``__unpack__()`` returns.
+
+Backwards Compatibility
+-----------------------
+
+Using the double asterisk syntax for annotating ``**kwargs`` would be available
+only in new versions of Python. :pep:`646` dealt with the similar problem and
+its authors introduced a new type operator ``Unpack``. For the purposes of this
+PEP, the proposition is to reuse ``Unpack`` for more precise ``**kwargs``
+typing. For example::
+
+    def foo(**kwargs: Unpack[Movie]) -> None: ...
+
+There are several reasons for reusing :pep:`646`'s ``Unpack``. Firstly, the
+name is quite suitable and intuitive for the ``**kwargs`` typing use case as
+the keywords arguments are "unpacked" from the ``TypedDict``. Secondly, there
+would be no need to introduce any new special forms. Lastly, the use of
+``Unpack`` for the purposes described in this PEP does not interfere with the
+use cases described in :pep:`646`.
+
+It is worth pointing out that the AST generated for ``Unpack`` would differ
```

Comment:
This seems unnecessary since this AST is standard for all Python 3 versions (Unpack isn't syntactically special) and we're not changing anything to it. I'd remove this paragraph.

Reply:
Right, makes sense, removed.